src/core/public/locale_helper.test.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { extractLocaleInfo, getAndUpdateLocaleInUrl } from './locale_helper';
+
+describe('extractLocaleInfo', () => {
+  const testCases = [
+    {
+      description: 'After hash and slash',
+      input: 'http://localhost:5603/app/home#/&i18n-locale=fr-FR',
```

Comment:
little bit worried about how long the url can get and as of late we have been able to hit the limit allowed. Is there another way we can indicate the locale without specifically calling it `i18n-locale`. I think I glanced it somewhere else but I think you called it out and makes sense why it is like this.

Because the i18n-locale is the template you provide with the value so it make sense. I've seen `g11n` be used for `globalization` by others to indicate it is the combination of both. is it worth it to do that?

Reply:
i think we also want to support dark theme through URL params, not sure about other settings through URL. If we add support for other settings might be good to have some consistent convention to convert setting name to URL parameter name

Reply:
HTTP headers is also another way to specify the locale. Can we support that?